File: articles/authority-legitimacy-institutions/go/authority_legitimacy_score.go
package main

import "fmt"

type LegitimacyCase struct {
	FormalAuthorityClarity float64
	ProceduralLegitimacy   float64
	OutcomeLegitimacy      float64
	Trust                  float64
	RuleClarity            float64
	SocialRecognition      float64
	Accountability         float64
	RepairCapacity         float64
	Fairness               float64
	ArbitrarinessPressure  float64
	VisibleInconsistency   float64
	UnequalBurden          float64
	OpacityPressure        float64
}

func AuthorityLegitimacyRaw(x LegitimacyCase) float64 {
	return 0.11*x.FormalAuthorityClarity +
		0.14*x.ProceduralLegitimacy +
		0.12*x.OutcomeLegitimacy +
		0.13*x.Trust +
		0.11*x.RuleClarity +
		0.11*x.SocialRecognition +
		0.12*x.Accountability +
		0.10*x.RepairCapacity +
		0.10*x.Fairness -
		0.14*x.ArbitrarinessPressure -
		0.10*x.VisibleInconsistency -
		0.09*x.UnequalBurden -
		0.08*x.OpacityPressure
}

func main() {
	demo := LegitimacyCase{82, 84, 78, 80, 82, 76, 79, 74, 81, 22, 20, 18, 19}
	fmt.Printf("Authority-legitimacy raw score: %.2f\n", AuthorityLegitimacyRaw(demo))
}

File: articles/authority-legitimacy-institutions/rust/authority_legitimacy_score.rs
struct LegitimacyCase {
    formal_authority_clarity: f64,
    procedural_legitimacy: f64,
    outcome_legitimacy: f64,
    trust: f64,
    rule_clarity: f64,
    social_recognition: f64,
    accountability: f64,
    repair_capacity: f64,
    fairness: f64,
    arbitrariness_pressure: f64,
    visible_inconsistency: f64,
    unequal_burden: f64,
    opacity_pressure: f64,
}

fn authority_legitimacy_raw(x: &LegitimacyCase) -> f64 {
    0.11 * x.formal_authority_clarity
        + 0.14 * x.procedural_legitimacy
        + 0.12 * x.outcome_legitimacy
        + 0.13 * x.trust
        + 0.11 * x.rule_clarity
        + 0.11 * x.social_recognition
        + 0.12 * x.accountability
        + 0.10 * x.repair_capacity
        + 0.10 * x.fairness
        - 0.14 * x.arbitrariness_pressure
        - 0.10 * x.visible_inconsistency
        - 0.09 * x.unequal_burden
        - 0.08 * x.opacity_pressure
}

fn main() {
    let demo = LegitimacyCase {
        formal_authority_clarity: 82.0,
        procedural_legitimacy: 84.0,
        outcome_legitimacy: 78.0,
        trust: 80.0,
        rule_clarity: 82.0,
        social_recognition: 76.0,
        accountability: 79.0,
        repair_capacity: 74.0,
        fairness: 81.0,
        arbitrariness_pressure: 22.0,
        visible_inconsistency: 20.0,
        unequal_burden: 18.0,
        opacity_pressure: 19.0,
    };

    println!(
        "Authority-legitimacy raw score: {:.2}",
        authority_legitimacy_raw(&demo)
    );
}
